stepcvt/cli/task.py

```diff
@@ -0,0 +1,59 @@
+import json
+from ..project import *
+from stepcvt import project
+
+
+def stlconvert(args):
+    # check for jsonfile
+    if args.jsonfile is None:
+        print("ERROR: Need to provide a jsonfile")
+        return 1
+
+    # try to get info from jsonfile
+    try:
+        with open(args.jsonfile, "r") as jf:
+            d = json.load(jf)
+            p = Project.from_dict(d)
+    except FileNotFoundError as fe:
+        print(f"ERROR: {args.jsonfile} doesn't exist")
+        return 1
+    except json.JSONDecodeError:
+        print("ERROR: Invalid json syntax")
+        return 1
+
+    if not p.sources:
+        print("ERROR: No existing source, add one first")
+    else:
+        source = p.sources[0]
+
+    if args.partID:
+        partID = args.partID
+    else:
+        print("ERROR: Need to specify at least one partid if --all is not included")
+        return 1
+
+    if args.rotation == True:
+        rotation = args.rotation
+    if args.linearTolerance == True:
+        linearTol = args.linearTolerance
+    else:
+        linearTol = 0.1
+    if args.angularTolerance == True:
+        angularTol = args.angularTolerance
+    else:
+        angularTol = 0.1
+
+    partID = project.STLConversionInfo(
+        rotation=rotation, linearTolerance=linearTol, angularTolerance=angularTol
+    )
+    partID.rotate(source)
+
+    with open(args.jsonfile, "w") as jf:
+        json.dump(partID.to_dict(), jf)
+    return 0
+
+
+# issues:
+# - there is no real way to connect a part to an STLConversionInfo object besides naming the object the partID
```

Comment:
I think you should first find the part in source using the `args.partid` and then either change the existing `STLConversionInfo` or create a new one.

Reply:
@ymh1003 are we storing a list of all the STLConversionInfo objects somewhere and if so, where?

Reply:
Each `partinfo` object has an attribute called `info`. This stores a list of `Taskinfo` objects, which may or may not contain the `STLConversionInfo` object.

Reply:
And source has partinfo objects so I could do source.contains(partID) and get the corresponding partinfo object for the ID?

Reply:
Oh I see - it is like this part of your code 
`for partid, obj in source.parts():
                    if partid in ids:
                        if partid in existing_ids:
                            print(f"{partid} has already been added")
                        else:
                            source.add_partinfo(partid, obj)`

Reply:
Right right.